content/en/docs/instrumentation/java/automatic/agent-config.md

```diff
@@ -173,6 +186,23 @@ span link connecting it to the producer trace.
 You can disable the agent using `-Dotel.javaagent.enabled=false` (or using the
 equivalent environment variable `OTEL_JAVAAGENT_ENABLED=false`).
 
+### Enable only specific instrumentation
+
+You can disable all default auto instrumentation and selectively re-enable individual instrumentation.
+This may be desireable to reduce startup overhead or to have more control of which instrumentation is applied.
+
+- Disable all instrumentation in the agent using `-Dotel.instrumentation.common.default-enabled=false`
+(or using the equivalent environment variable `OTEL_INSTRUMENTATION_COMMON_DEFAULT_ENABLED=false`).
+- Enable each desired instrumentation individually using `-Dotel.instrumentation.[name].enabled=true`
+(or using the equivalent environment variable `OTEL_INSTRUMENTATION_[NAME]_ENABLED`) where `name`
+(`NAME`) is the corresponding instrumentation `name` below.
+
+### Enable manual instrumentation only
+
+You can suppress all auto instrumentations but have support for manual
+instrumentation with `@WithSpan` and normal API interactions by using
+`-Dotel.instrumentation.common.default-enabled=false -Dotel.instrumentation.opentelemetry-api.enabled=true -Dotel.instrumentation.opentelemetry-instrumentation-annotations.enabled=true`
```

Comment:
Interesting, so the other two are also considered off if you disable `default-enabled`?

Reply:
There are a small subset of integrations that don't follow the "default enabled" flag by manually returning a boolean.  ([For example](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/internal/internal-url-class-loader/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/urlclassloader/UrlClassLoaderInstrumentationModule.java#L23-L26).) 

Otherwise, everything else follows the default config value.

Reply:
See https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7201 we'll probably change it so that these wouldn't be affected by `default-enabled`